test/metabase/query_processor_test/date_time_zone_functions_test.clj

```diff
@@ -169,3 +177,141 @@
+                     {:expressions {"expr" [op [:field field-id nil] 2 unit]}
+                      :fields      [[:expression "expr"]]}]
+
+                    [(into [] (frequencies
+                                [(date-math op #t "2004-03-19 09:19:09" 2 unit col-type) (date-math op #t "2008-06-20 10:20:10" 2 unit col-type)
+                                 (date-math op #t "2012-11-21 11:21:11" 2 unit col-type) (date-math op #t "2012-11-21 11:21:11" 2 unit col-type)]))
+                     {:expressions {"expr" [op [:field field-id nil] 2 unit]}
+                      :aggregation [[:count]]
+                      :breakout    [[:expression "expr"]]}]]]
+             (testing (format "%s %s function works as expected on %s column for driver %s" op unit col-type driver/*driver*)
+               (is (= (set expected) (set (test-date-extract query))))))))
+
+       (testing "date arithmetic with date columns"
+         (let [[col-type field-id] [:date (mt/id :times :d)]]
+           (doseq [op               [:date-add :date-subtract]
+                   unit             [:year :quarter :month :day]
+                   [expected query]
+                   [[[[(date-math op #t "2004-03-19 00:00:00" 2 unit col-type)] [(date-math op #t "2008-06-20 00:00:00" 2 unit col-type)]
+                      [(date-math op #t "2012-11-21 00:00:00" 2 unit col-type)] [(date-math op #t "2012-11-21 00:00:00" 2 unit col-type)]]
+                     {:expressions {"expr" [op [:field field-id nil] 2 unit]}
+                      :fields      [[:expression "expr"]]}]
+
+                    [(into [] (frequencies
+                                [(date-math op #t "2004-03-19 00:00:00" 2 unit col-type) (date-math op #t "2008-06-20 00:00:00" 2 unit col-type)
+                                 (date-math op #t "2012-11-21 00:00:00" 2 unit col-type) (date-math op #t "2012-11-21 00:00:00" 2 unit col-type)]))
+                     {:expressions {"expr" [op [:field field-id nil] 2 unit]}
+                      :aggregation [[:count]]
+                      :breakout    [[:expression "expr"]]}]]]
+             (testing (format "%s %s function works as expected on %s column for driver %s" op unit col-type driver/*driver*)
+               (is (= (set expected) (set (test-date-extract query))))))))
+
+      (when-not (> (mongo-major-version (mt/db)) 4)
+        (doseq [op [:date-add :date-subtract]]
+          (is (thrown-with-msg? clojure.lang.ExceptionInfo #"Date arithmetic not supported in versions before 5"
+                                (mt/compile (mt/mbql-query times {:expressions {"expr" [op [:field (mt/id :times :dt) nil] 2 :year]}
+                                                                  :fields      [[:expression "expr"]]}))))))))))
+
+(deftest date-math-with-extract-test
+  (mt/test-drivers (disj (mt/normal-drivers-with-feature :date-arithmetics) :mongo)
```

Comment:
If you're excluding a driver you should leave a comment with a note as to why you are doing so. If it's because there is a different version of the test for mongo then include a `[[link]]` to it in your comment